avocado/plugins/multiplex.py

```diff
@@ -35,12 +35,10 @@ def __init__(self, *args, **kwargs):
         super(Multiplex, self).__init__(*args, **kwargs)
 
     def configure(self, parser):
-        if multiplexer.MULTIPLEX_CAPABLE is False:
-            return
-
         parser = super(Multiplex, self).configure(parser)
-        parser.add_argument('multiplex', nargs='+',
-                            help='Path(s) to a multiplex file(s)')
+        if multiplexer.MULTIPLEX_CAPABLE:
+            parser.add_argument("-m", '--multiplex', nargs='*',
```

Comment:
Actual the problem is a bit simpler. In the RFC we decided we don't want to turn the `multiplexer` into a plugin yet (maybe in the future), we only want to define the API to feed it and turn the yaml parser into a plugin. This should allow not just switching between implementations, but it should allow coexisting of different plugins at the same time. So a valid usage would be:

```
avocado multiplex --multiplex file.yaml --cartesian-config base.cfg --ini samlpe.ini
```

which would merge all the results and show the product. So we could either say `--multiplex` is special and keep it as a positional argument, or we can avoid it by using a different argument.

Also note that in the WiP version I asked, whether it'd be OK to do some renaming and I think it'd help understanding the change. Ideally I'd do:

```
--multiplex|-m => --mux-yaml|-m
avocado.core.multiplexer => mux
avocado.plugins.multiplex => avocado.plugins.yaml_to_mux (or mux_yaml`)
```

This change would make it clearer as `Mux` is a params database including the variants generator and `mux_yaml` is a plugin to parse yaml into a params database.

I haven't change the argument as I was afraid people depend on it heavily. But I think it'd make things clearer.